Tools/buildstream/patches/fdo-0009-gst-plugins-bad-Enable-x265-encoder.patch

```diff
@@ -0,0 +1,405 @@
++      mv "/buildstream-build/source/_builddir/libx265.so.199" "%{install-root}%{libdir}"
++      mkdir -p "%{install-root}%{libdir}/pkgconfig"
++      cp ../x265.pc "%{install-root}%{libdir}/pkgconfig/"
++
++public:
++  bst:
++    split-rules:
++      devel:
++        (>):
++        - '%{libdir}/lib*.so'
++        - '%{libdir}/pkgconfig'
++        - '%{libdir}/pkgconfig/*.pc'
++        - '%{includedir}'
++        - '%{includedir}/x265.h'
++        - '%{includedir}/x265_config.h'
++        - '%{includedir}/hdr10plus.h'
++
++(@):
++- elements/include/x265-source.yml
+diff --git a/elements/include/x265-source.yml b/elements/include/x265-source.yml
+new file mode 100644
+index 000000000..e75643a75
+--- /dev/null
++++ b/elements/include/x265-source.yml
+@@ -0,0 +1,17 @@
++sources:
++- kind: tar
++  url: https://bitbucket.org/multicoreware/x265_git/downloads/x265_3.5.tar.gz
++  ref: e70a3335cacacbba0b3a20ec6fecd6783932288ebc8163ad74bcc9606477cae8
++# https://github.com/rpmfusion/x265/
++- kind: patch
++  path: patches/x265/x265-pic.patch
++- kind: patch
++  path: patches/x265/x265-high-bit-depth-soname.patch
++- kind: patch
++  path: patches/x265/x265-detect_cpu_armhfp.patch
++- kind: patch
++  path: patches/x265/x265-arm-cflags.patch
++- kind: patch
++  path: patches/x265/x265-pkgconfig_path_fix.patch
++- kind: local
++  path: files/x265.pc
+diff --git a/files/x265.pc b/files/x265.pc
+new file mode 100644
+index 000000000..a5fb43463
+--- /dev/null
++++ b/files/x265.pc
+@@ -0,0 +1,11 @@
++prefix=/usr
++exec_prefix=${prefix}
++libdir=/usr/lib/x86_64-linux-gnu
++includedir=${prefix}/include
++
++Name: x265
++Description: H.265/HEVC video encoder
++Version: 3.5
++Libs: -L${libdir} -lx265
++Libs.private: -lstdc++ -lm -lgcc_s -lgcc -lgcc_s -lgcc -lrt -ldl -lnuma
++Cflags: -I${includedir}
+diff --git a/patches/x265/x265-arm-cflags.patch b/patches/x265/x265-arm-cflags.patch
+new file mode 100644
+index 000000000..8cd5c867d
+--- /dev/null
++++ b/patches/x265/x265-arm-cflags.patch
+@@ -0,0 +1,63 @@
++--- x265_3.4/source/CMakeLists.txt.cflags
+++++ x265_3.4/source/CMakeLists.txt
++@@ -238,28 +238,6 @@
++             endif()
++         endif()
++     endif()
++-    if(ARM AND CROSS_COMPILE_ARM)
++-        if(ARM64)
++-            set(ARM_ARGS -fPIC)
++-        else()
++-            set(ARM_ARGS -march=armv6 -mfloat-abi=soft -mfpu=vfp -marm -fPIC)
++-        endif()
++-        message(STATUS "cross compile arm")
++-    elseif(ARM)
++-        if(ARM64)
++-            set(ARM_ARGS -fPIC)
++-            add_definitions(-DHAVE_NEON)
++-        else()
++-            find_package(Neon)
++-            if(CPU_HAS_NEON)
++-                set(ARM_ARGS -mcpu=native -mfloat-abi=hard -mfpu=neon -marm -fPIC)
++-                add_definitions(-DHAVE_NEON)
++-            else()
++-                set(ARM_ARGS -mcpu=native -mfloat-abi=hard -mfpu=vfp -marm)
++-            endif()
++-        endif()
```

Comment:
This logic doesn't make much sense to me.

1. Why making a difference between the flags passed to the compiler when cross-compiling and when not?. We do care about redistributable binaries on the SDK. So we should always build with compatible flags to the minimum baseline supported at run-time and not use as baseline whatever the CPU that builds the SDK happens to have. So I suggest to avoid using flags like `-mcpu=native`
2. Why use `march=armv6 -mfloat-abi=soft` for ARMv7 when cross-compiling but end using `-mcpu=native -mfloat-abi=hard` when building natively for ARMv7?

Reply:
I copied those patches from https://github.com/rpmfusion/x265/
I can remove the ones related with armv6/7 if needed, we don't support those legacy arches anyway.

Reply:
ok, is fine I guess. If in the end we end supporting armv7 this needs an update but right now we don't support it.